engines/zypper_auth/lib/zypper_auth/engine.rb

```diff
@@ -51,9 +51,22 @@ def verify_instance(request, logger, system, params_product_id = nil)
     end
 
     def handle_scc_subscription(request, system, verification_provider, params_product_id = nil)
-      product_class = Product.find_by(id: params_product_id).product_class if params_product_id.present?
-      result = SccProxy.scc_check_subscription_expiration(request.headers, system, product_class)
-      return true if result[:is_active]
+      if params_product_id.present?
+        product_class = Product.find_by(id: params_product_id).product_class
+        result = SccProxy.scc_check_subscription_expiration(request.headers, system, product_class)
+        return true if result[:is_active]
```

Comment:
If `result[:is_active]` is not true here, the method would return the value from the line before, which is `result`. I suggest to always return true or false here: 

```suggestion
        return result[:is_active]
```